src/toxcore/dht/dht_node.rs

```diff
@@ -30,30 +53,24 @@ If both node is Good node, then we compare PK's distance.
 */
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct DhtNode {
-    /// Socket addr of node for IPv4.
-    pub saddr_v4: Option<SocketAddrV4>,
-    /// Socket addr of node for IPv6.
-    pub saddr_v6: Option<SocketAddrV6>,
+    /// Socket addr and times of node for IPv4.
+    pub assoc4: SockAndTime,
```

Comment:
Again, `pub saddr: Option<SocketAddr>` from SockAndTime may keep IPv6 (from the point of view of the type system).